File: Javascript/Hashtables/find_first_recurring_character.js
function find_first_recurring_character(arr){
    // Function to find first recurring character in an array

    // Store each character in the hashtable and if we come across a character
    // already in the hashtable we return the item
    hashTable = {};

    for(let item of arr){
        if(hashTable[item]){
            return item;
        }
        hashTable[item] = 1;
    }

    return "No recurring items"
}

arr = [1,2,3,4]
console.log(find_first_recurring_character(arr))

File: Python/Data Structures/Hashtables/find_first_recurring_character.py
def find_first_recurring_character(arr):
    """ Function to find first recurring character in an array """

    hashtable = {}
    for item in arr:
        if hashtable.get(item, None) is not None:
            return item
        hashtable[item] = 1

    return "No recurring item"


arr = [1, 2, 3, 4, 5, 6, 3]
print(find_first_recurring_character(arr))
